### PR TITLE
Pie 2721/sort notifications

### LINKED ISSUE #2721: Sort notifications by most recent first in the backend
## 📋 Summary

Using the data format that has already been agreed upon, sort notifications so that the newest notifications show up at the top. 
For example:
notification 1: Eleanor Pena’s subsidy authorization expires on May 31st. Email the updated letter to team@pieforproviders.com
May 1, 2022
notification 2: Jenny Wilson’s subsidy authorization expires on July 15th. Email the updated letter to team@pieforproviders.com
Jun 15, 2022

notification 2 should be sorted first
then notification 1

## 🇺🇸 States this change applies to
- [X] ALL
- [ ] Nebraska
- [ ] Illinois

### PR DESCRIPTION
**Pull request by OliverCai0**
## 💅🏼 What issue does this fix?
Closes #2721, which requests that notifications be ordered chronologically based on their expiry date before they are sent. 

## 🧳 Steps to test
Added additional tests to spec/requests/api/v1/notifications_spec.rb

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,7 +4,6 @@
 # or operating system, you probably want to add a global ignore instead:
 #   git config --global core.excludesfile '~/.gitignore_global'
 
-# Mac OS
```
Comment (katelovescode) [inline .gitignore:7]: @OliverCai0 can you remove these changes from this PR?  I don't think they're necessary

File: db/schema.rb [hunk 2]
```diff
@@ -13,11 +13,10 @@
 ActiveRecord::Schema.define(version: 2022_06_08_035549) do
 
   # These are extensions that must be enabled in order to support this database
-  enable_extension "pg_stat_statements"
```
Comment (katelovescode) [inline db/schema.rb:16]: Also please remove this file from your PR - you didn't change the db for this feature so we shouldn't commit unrelated changes

File: spec/requests/api/v1/notifications_spec.rb [hunk 3]
```diff
@@ -6,12 +6,21 @@
...
+  let!(:early_approval) { create(:approval, num_children: 1, business: business, expires_on: 1.day.after) }
+  let!(:late_approval) { create(:approval, num_children: 1, business: business, expires_on: 29.days.after) }
   let!(:children) { approval.children }
+  let!(:early_child) { early_approval.children.first }
+  let!(:late_child) { late_approval.children.first }
   let!(:non_owner_child) { create(:necc_child) }
 
   before do
-    create(:notification, child: children.first, approval: approval)
+    approvals = Array.new(3) { |_i| approval }
```
Comment (katelovescode) [inline spec/requests/api/v1/notifications_spec.rb:18]: I'm a little confused about this configuration/setup, can you explain what your goal is here?

**Review by katelovescode — CHANGES_REQUESTED** [review]
Just a few cleanup requests in the code - also please remove `pie_erd` since that's not related to your change either.  The only things that should be in this PR are the controller and the spec.

File: spec/requests/api/v1/notifications_spec.rb [hunk 4]
```diff
@@ -6,12 +6,17 @@
...
+  let!(:early_approval) { create(:approval, num_children: 1, business: business, expires_on: 1.day.after) }
+  let!(:late_approval) { create(:approval, num_children: 1, business: business, expires_on: 29.days.after) }
+  let!(:children) { [approval.children.first, early_child, late_child] }
+  let!(:early_child) { early_approval.children.first }
+  let!(:late_child) { late_approval.children.first }
   let!(:non_owner_child) { create(:necc_child) }
 
   before do
-    create(:notification, child: children.first, approval: approval)
+    approvals = [approval, early_approval, late_approval]
```
Comment (OliverCai0) [inline spec/requests/api/v1/notifications_spec.rb:18]: This creates three notifications with three different approval expiry times that are intentionally in the wrong order.

**Review by katelovescode — APPROVED** [review]

**Review by katelovescode — APPROVED** [review]

**Comment by katelovescode** [comment]
Closing in favor of the rebase b/c this won't merge

**Pull request closed without merging**